src/components/common/item-selector/chickleted-input.js

```diff
@@ -68,10 +68,11 @@ const ChickletTag = styled.span`
   }
 `;
 
-const Chicklet = ({disabled, name, remove}) => (
+const Chicklet = ({disabled, name, remove, addon}) => (
   <ChickletButton>
     <ChickletTag>{name}</ChickletTag>
-    <Delete height="10px" onClick={disabled ? null : remove} />
+    {Boolean(addon) && addon(name)}
+    <Delete onClick={disabled ? null : remove} />
```

Comment:
instead of passing a `addon` to on the chicklet, you can create a new prop `ChickletIButtonRenderComponent` to item-selector. just like the `DropDownLineItemRenderComponent`. And default value would be this `ChickletButton`

Reply:
@heshan0131  Default? It's only needed in tooltip scenario.

Reply:
Duplicate whole Chicklet component w/ tooltip config dropdown